metadata-ingestion/src/datahub/ingestion/source/cassandra.py

```diff
@@ -0,0 +1,413 @@
+
+# This class helps convert cassandra column types to SchemaFieldDataType for use by the datahaub metadata schema
+class CassandraToSchemaFieldConverter:
+    # FieldPath format version.
+    version_string: str = "[version=2.0]"
+
+    # Mapping from cassandra field types to SchemaFieldDataType.
+    # https://cassandra.apache.org/doc/stable/cassandra/cql/types.html (version 4.1)
+    _field_type_to_schema_field_type: Dict[str, Type] = {
+        # Bool
+        "boolean": BooleanTypeClass,
+        # Binary
+        "blob": BytesTypeClass,
+        # Numbers
+        "bigint": NumberTypeClass,
+        "counter": NumberTypeClass,
+        "decimal": NumberTypeClass,
+        "double": NumberTypeClass,
+        "float": NumberTypeClass,
+        "int": NumberTypeClass,
+        "smallint": NumberTypeClass,
+        "tinyint": NumberTypeClass,
+        "varint": NumberTypeClass,
+
+        # Dates
+        "date": DateTypeClass,
+
+        # Times
+        "duration": TimeTypeClass,
+        "time": TimeTypeClass,
+        "timestamp": TimeTypeClass,
+        
+        # Strings
+        "text": StringTypeClass,
+        "ascii": StringTypeClass,
+        "inet": StringTypeClass,
+        "timeuuid": StringTypeClass,
+        "uuid": StringTypeClass,
+        "varchar": StringTypeClass,
+
+        # Records
+        "geo_point": RecordTypeClass,
+
+        # Arrays
+        "histogram": ArrayTypeClass,
+    }
+
+    @staticmethod
+    def get_column_type(cassandra_column_type: str) -> SchemaFieldDataType:
+        type_class: Optional[
+            Type
+        ] = CassandraToSchemaFieldConverter._field_type_to_schema_field_type.get(
+            cassandra_column_type
+        )
+        if type_class is None:
+            logger.warning(
+                f"Cannot map {cassandra_column_type!r} to SchemaFieldDataType, using NullTypeClass."
+            )
+            type_class = NullTypeClass
+
+        return SchemaFieldDataType(type=type_class())
+
+    def __init__(self) -> None:
+        self._prefix_name_stack: List[str] = [self.version_string]
+
+    def _get_cur_field_path(self) -> str:
+        return ".".join(self._prefix_name_stack)
+
+    def _get_schema_fields(
+        self, cassandra_column_schemas: [dict[str, any]]
+    ) -> Generator[SchemaField, None, None]:
+        # append each schema field (sort so output is consistent)
+        for columnSchema in cassandra_column_schemas:
+            columnName: str =  getattr(columnSchema, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["column_name"])
+            cassandra_type: str =  getattr(columnSchema, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["column_type"])
+            if cassandra_type is not None:
+                self._prefix_name_stack.append(f"[type={cassandra_type}].{columnName}")
+                schema_field_data_type = self.get_column_type(cassandra_type)
+                schema_field = SchemaField(
+                    fieldPath=self._get_cur_field_path(),
```

Comment:
just so i understand this correctly, this path is prefixed by the version of `SchemaField`?